File: packages/Client/sdk/models/CampaignAllocationEntry.ts
import { CampaignAllocation, Model, model } from 'sdk'

export const enum CampaignAllocationEntryType { Fund, Refund }

@model('CampaignAllocationEntry')
export class CampaignAllocationEntry extends Model {
	allocationId!: number
	allocation!: CampaignAllocation
	type!: CampaignAllocationEntryType
	paymentMethodIdentifier!: string
	data!: string
	amount!: number
}

File: packages/Server/Application/Campaigns/Allocation/CampaignAllocationEntry.cs
namespace Solidarity.Application.Campaigns.Allocation;

public enum CampaignAllocationEntryType { Fund, Refund }

public sealed class CampaignAllocationEntry : Entity
{
	public int CampaignAllocationId { get; set; }
	public CampaignAllocation CampaignAllocation { get; set; } = null!;
	public CampaignAllocationEntryType Type { get; set; }
	public string PaymentMethodIdentifier { get; set; } = null!;
	public string Data { get; set; } = null!;
	public decimal Amount { get; set; }
}